### PR TITLE
Enable formatOnType feature

### PR DESCRIPTION
**Pull request by kapilmb**
This enables "editor.formatOnType" option in vscode settings.

Related PR: PowerShell/PowerShellEditorServices#370

### CODE REVIEW

**Comment by kapilmb** [comment]
This PR is dependent on changes in #501 and hence should not be merged before #501 is merged.

**Comment by kapilmb** [comment]
Needs rebase after #501 is merged.

File: src/features/DocumentFormatter.ts [hunk 1]
```diff
@@ -384,6 +467,11 @@ export class DocumentFormatterFeature implements IFeature {
         this.rangeFormattingEditProvider = vscode.languages.registerDocumentRangeFormattingEditProvider(
             "powershell",
             this.documentFormattingEditProvider);
+        this.onTypeFormattingEditProvider = vscode.languages.registerOnTypeFormattingEditProvider(
+            "powershell",
+            this.documentFormattingEditProvider,
+            this.firstTriggerCharacter,
+            ...this.moreTriggerCharacters);
```
Comment (daviwil) [inline src/features/DocumentFormatter.ts]: This is such a bizarre way for them to define this API...  If it's a new API why have a "first" param and then a "more" param?  Weird.

Comment (kapilmb) [inline src/features/DocumentFormatter.ts]: Yes, they could just have one argument taking in triggerCharacters instead of this!

**Comment by kapilmb** [comment]
Rebased on `develop` after #501 merge and removed some redundant code.

**Pull request merged**